pkg/informer/event.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2020 Red Hat, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package informer
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/informers/internalinterfaces"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	ksmwatch "k8s.io/kube-state-metrics/pkg/watch"
```

Comment:
Any reason why you choose to import this vs just recreate these functions, you need to rewrite them anyways to match the options?

Reply:
I chose to import it because I wanted to use existing code to instrument the informers and the ksm watch package was fairly easy to integrate.
But you are very right that it is not abstracting much, I only gain the instrumented List/Watch functions from it.

Considering that, it might not be worthwhile to import ksm and I should just rewrite the package here.